tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,112 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests_test
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
+)
+
+var _ = Describe("macvtap-cni", func() {
+
+	Describe("macvtap resource creation", func() {
+
+		Context("a lower device is exposed via a ConfigMap", func() {
+			configMapName := "macvtap-dp-config"
+			lowerDevice := "eth0"
+			quantity := 50
+			namespace := "default"
+
+			BeforeEach(func() {
+				macvtapConfig := &v1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "",
+						Name:      configMapName,
+					},
+					Data: map[string]string{
+						"DP_MACVTAP_CONF": fmt.Sprintf(`
+							[ 
+								{
+        							"name" : %s,
+									"master" : %s,
+									"mode": "bridge",
+									"capacity" : %d
+						 		}
+							]`, lowerDevice, lowerDevice, quantity),
+					},
+				}
+				_, err := clientset.CoreV1().ConfigMaps(namespace).Create(macvtapConfig)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				err := clientset.CoreV1().ConfigMaps(namespace).Delete(configMapName, nil)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			Context("macvtap-cni device plugin operates accordingly", func() {
+				It("reports the lower device as available", func() {
+					expectedResourceName := v1.ResourceName(fmt.Sprintf("macvtap.network.kubevirt.io/%s", lowerDevice))
+					nodes, _ := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+					for _, node := range nodes.Items {
+						expectedQuantity, err := resource.ParseQuantity(strconv.Itoa(quantity))
+						Expect(err).NotTo(HaveOccurred())
+
+						// confirm capacity is OK
+						Expect(node.Status.Capacity).Should(HaveKey(expectedResourceName))
+						macvtapCapacity := node.Status.Capacity[v1.ResourceName(expectedResourceName)]
+						Expect(macvtapCapacity).To(Equal(expectedQuantity))
+
+						// confirm allocatable is OK
+						Expect(node.Status.Allocatable).Should(HaveKey(expectedResourceName))
+						macvtapAllocatable := node.Status.Allocatable[v1.ResourceName(expectedResourceName)]
+						Expect(macvtapAllocatable).To(Equal(expectedQuantity))
+					}
+				})
+			})
+
+			Context("a VM is created with direct connectivity to the lower device", func() {
+				BeforeEach(func() {
+
+				})
+
+				AfterEach(func() {})
+
+				It("successfully reaches the internet", func() {
+
+				})
+			})
+
+			Context("VMs are created with direct connectivity to the lower device", func() {
+				It("the VMs can successfully ping each other via the macvtap interface", func() {
+
+				})
+
+				It("the VMs cannot ping the host where they are co-located", func() {
+
+				})
+
+				It("the VMs can ping the host where they are not located", func() {
+
+				})
```

Comment:
Additional context not needed I think. `"that can be used to reach another VM"`, `"that can be used to reach a remote host"`, `"that cannot be used to reach the local host"`. Also, if the specs are not ready to run, I read you can put a P in front to prevent it: `PIt(...)`

Reply:
Done.

Tks for 'pending' tests pointer, I was unaware of it.